### DISCUSSION TITLE
Inquiry about the frequency of data updates in the jolpica-f1 API

### DISCUSSION THREAD

**Discussion by ijro** [jolpica/jolpica-f1#95] [Q&A] [answered]
First of all, I would like to express my appreciation for the amazing work you’ve done with the jolpica-f1 API. It’s a fantastic tool for Formula 1 fans and developers, and your dedication to providing this resource to the community is truly valued.

I’m currently studying and utilizing your API for a project and was hoping to get some clarification regarding the _update frequency_ of the race results and data. Could you kindly let me know how often the data is refreshed after a race or event? Understanding this would greatly help me with my application.

Thank you once again for your hard work and for making this resource available to us all.

**Comment by jolpica** [comment, marked as answer]
For this season we are aiming to do a single update per race weekend on the Monday after the event.
However, there may be some delays outside of our control while we are using ergast images while we refine our import scripts.

By next season we'll be doing at least 1 update per race on the Monday, but Ideally we'd like to update the data a few hours after each session, but no promises on that right now.

**Reply by williamsdb** [reply]
I agree with @ijro 💯. Great to have a replacement for Ergast.

Regarding data updates when do you think the static data will be available each season? By which I mean the drivers, races, constructors?

**Reply by MonsterBandit** [reply]
seconded

**Reply by CarbonatedCarbon** [reply]
thirded

**Reply by jolpica** [reply]
This season, Hopefully by end of this month at the latest (I've been quite busy sorry!).

Going forward, the data should be added relatively quickly after becoming available once we've built up our data collection and I'm no longer as much of a bottleneck.

**Reply by williamsdb** [reply]
No problem at all I'm just grateful to have a replacement!

**Comment by Oldertarl** [comment]
Happy to help and assist where possible


Op ma 6 jan 2025 om 12:08 schreef Jess Perry ***@***.***>:

> This season, Hopefully by end of this month at the latest (I've been quite
> busy sorry!).
>
> Going forward, the data should be added relatively quickly after becoming
> available once we've built up our data collection and I'm no longer as much
> of a bottleneck.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jolpica/jolpica-f1/discussions/95#discussioncomment-11747538>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAGFWGDTKJPHKWLUPGBJXFD2JJP4LAVCNFSM6AAAAABQKG7IKKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNZUG42TGOA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 

Best regards
Fons Seesink

**Comment by jolpica** [comment]
Update on this, drivers, teams & race schedule (Only dates) should be available now

https://api.jolpi.ca/ergast/f1/2025/drivers/?limit=20
https://api.jolpi.ca/ergast/f1/2025/races/

Drivers/teams before the season starts wasn't possibe previously on ergast, so hopefully features like this will make the initial delays worth it!

**Reply by williamsdb** [reply]
Perfect, thanks, I'll give it a go.

**Reply by williamsdb** [reply]
Hi

Just had a look at the races data and it is missing the following:

```
"FirstPractice": {
                        "date": "2024-02-29",
                        "time": "11:30:00Z"
                    },
                    "SecondPractice": {
                        "date": "2024-02-29",
                        "time": "15:00:00Z"
                    },
                    "ThirdPractice": {
                        "date": "2024-03-01",
                        "time": "12:30:00Z"
                    },
                    "Qualifying": {
                        "date": "2024-03-01",
                        "time": "16:00:00Z"
                    }
```

Is this because it is yet to be added, is no longer supported or something else?

I ask as it affects what I do in my code.

Thanks!

**Reply by jolpica** [reply]
Its unavailable as those times have not yet been confirmed 

https://www.formula1.com/en/racing/2025/australia

They will be added after we have some confirmation. Sorry for the inconvenience.
